packages/checkbox-group/src/LionCheckboxIndeterminate.js

```diff
@@ -0,0 +1,110 @@
+import { html, css } from '@lion/core';
+import { LionCheckbox } from './LionCheckbox.js';
+
+export class LionCheckboxIndeterminate extends LionCheckbox {
+  static get properties() {
+    return {
+      /**
+       * Indeterminate state of the checkbox
+       */
+      indeterminate: {
+        type: Boolean,
+        reflect: true,
+      },
+    };
+  }
+
+  get _checkboxGroupNode() {
+    return /** @type {import('./LionCheckboxGroup').LionCheckboxGroup} */ (this.parentElement);
+  }
+
+  get _subCheckboxes() {
+    return this._checkboxGroupNode.formElements.filter(checkbox => checkbox !== this);
+  }
+
+  _parentModelValueChanged() {
+    const checkedElements = this._subCheckboxes.filter(checkbox => checkbox.checked);
+    switch (this._subCheckboxes.length - checkedElements.length) {
+      // all checked
+      case 0:
+        this.indeterminate = false;
+        this.checked = true;
+        break;
+      // none checked
+      case this._subCheckboxes.length:
+        this.indeterminate = false;
+        this.checked = false;
+        break;
+      default:
+        this.indeterminate = true;
+    }
+  }
+
+  _ownInputChanged() {
+    this._subCheckboxes.forEach(checkbox => {
+      // eslint-disable-next-line no-param-reassign
+      checkbox.checked = this._inputNode.checked;
+    });
+  }
+
+  /**
+   * @override
+   * clicking on indeterminate status will set the status as checked
+   */
+  // eslint-disable-next-line class-methods-use-this
+  __toggleChecked() {}
+
+  // eslint-disable-next-line class-methods-use-this
+  _afterTemplate() {
+    return html`
+    <div class"nestedCheckboxes">
+      <slot></slot>
+    </div>`;
+  }
+
+  constructor() {
+    super();
+    this.indeterminate = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._parentModelValueChanged = this._parentModelValueChanged.bind(this);
+    this._checkboxGroupNode.addEventListener('model-value-changed', this._parentModelValueChanged);
+    this._ownInputChanged = this._ownInputChanged.bind(this);
+    this._inputNode.addEventListener('change', this._ownInputChanged.bind(this));
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this._checkboxGroupNode.removeEventListener(
+      'model-value-changed',
+      this._parentModelValueChanged,
+    );
+    this._inputNode.removeEventListener('change', this._ownInputChanged);
+  }
+
+  /** @param {import('lit-element').PropertyValues } changedProperties */
+  updated(changedProperties) {
+    super.updated(changedProperties);
+    if (changedProperties.has('indeterminate')) {
+      this._inputNode.indeterminate = this.indeterminate;
+    }
+  }
+
+  static get styles() {
+    const superCtor = /** @type {typeof LionCheckbox} */ (super.prototype.constructor);
+    return [
+      superCtor.styles ? superCtor.styles : [],
+      css`
+        :host .nestedCheckboxes {
+          display: block;
+        }
+
+        :host ::slotted(lion-checkbox) {
```

Comment:
If someone would extend LionCheckbox (for instance call it `wolf-checkbox`) this selector would break.
I think `[slot=input]` will be a safe choice

Reply:
I think it's using an unnamed slot in the afterTemplate, but I agree it would be better to name it. Not sure what the best name for it is though, but `input` is for sure already taken by the native <input type="checkbox"> so perhaps it should be `after` or `content`

Reply:
Or slot="checkbox", that makes more sense 👍

Reply:
Yes indeed, I was confused. This is actually something we cannot style, since we don't know the element name and are not sure that it's a direct child of the default slot (would be possible to wrap this in `ul > li` for instance.
Maybe add a margin of 8px  to `.choice-field__nested-checkboxes` and delete this selector?

Reply:
Ah true, even if you put [slot="checkbox"] this will only apply to direct children, and could be a <div> in between or some other html structures. I agree, might be better to apply the styling to the container in the shadow root 👍

Reply:
Hm but hold on, what if you do

```html
<lion-checkbox-indeterminate label="Scientists">
  <ul slot="checkbox">
    <lion-checkbox label="Isaac Newton" .choiceValue=${'Isaac Newton'}></lion-checkbox>
    <lion-checkbox label="Galileo Galilei" .choiceValue=${'Galileo Galilei'}></lion-checkbox>
  </ul>
</lion-checkbox-indeterminate>
```

That works fine actually, also with the ::slotted([slot="checkbox"]) selector. 

What doesn't work is:

```html
<lion-checkbox-indeterminate label="Scientists">
  <ul>
    <lion-checkbox slot="checkbox" label="Isaac Newton" .choiceValue=${'Isaac Newton'}></lion-checkbox>
    <lion-checkbox slot="checkbox" label="Galileo Galilei" .choiceValue=${'Galileo Galilei'}></lion-checkbox>
  </ul>
</lion-checkbox-indeterminate>
```

But this doesn't work from HTML perspective already (apart from the fact that the CSS ::slotted doesn't work anymore as well).

So perhaps the current way is fine? Although slightly "odd" to put a <ul> in a slot that is called "checkbox", so then maybe the name should be a little more generic or we make it a default slot without a name? I am not usually a fan of unnamed slots but here it might make sense, as you can essentially put whatever you want in it, and the only result is that you get some left padding.

Reply:
A `<ul>` needs `<li>`'s. You probably want to avoid that.

Reply:
True, maybe a bad example, `<div>` could happen though e.g. if someone needs a wrapper around it with a border or something, for their design system.

Reply:
Or `<ul slot="checkboxes"><li><ing-checkbox>` (something similar is used in MDN and w3c examples).
But I'm not a fan of a named slot in this particular case, unnamed results in best DX imho.
It's also more in line with the pattern we follow for a regular fieldset/checkbox-group/radio-group etc.